### PR TITLE
gixpp (gixsql) speedup: prevent excessive filename conversions to absolute path

### PR DESCRIPTION
**Pull request by GitMensch**
Seen when probing with GDB "early" in the preprocessing - GDB was often somewhere in the C++ routines for this call.
Warning: I _think_ the code works the same, just with many less system calls and intermediate objects - but as I don't know what those functions actually are used for (comments on function scope would help) I do not know if this is the correct patch.

### CODE REVIEW

**Comment by GitMensch** [comment]
merge-conflicts fixed, ping @mridoni for possible integration

**Comment by GitMensch** [comment]
Any comments on this?

File: libgixpp/TPESQLProcessing.cpp [hunk 1]
```diff
@@ -342,7 +343,7 @@ bool TPESQLProcessing::processNextFile()
 
 		bool in_ws = (input_line >= working_begin_line) && (input_line <= working_end_line);
 
-		cb_exec_sql_stmt_ptr exec_sql_stmt = find_exec_sql_stmt(the_file, input_line);
+		cb_exec_sql_stmt_ptr exec_sql_stmt = find_exec_sql_stmt(f1, input_line);
```
Comment (GitMensch) [inline libgixpp/TPESQLProcessing.cpp:346]: The issue is that this is called for each line...

File: libgixpp/TPESQLProcessing.cpp [hunk 2]
```diff
@@ -564,12 +565,11 @@ bool TPESQLProcessing::put_call(const ESQLCall &c, bool terminate_with_period)
...
 }
 
-cb_exec_sql_stmt_ptr TPESQLProcessing::find_exec_sql_stmt(const std::string filename, int i)
+cb_exec_sql_stmt_ptr TPESQLProcessing::find_exec_sql_stmt(const std::string f1, int i)
 {
 	std::vector<cb_exec_sql_stmt_ptr> *p = main_module_driver.exec_list;
 	for (auto e : *p) {
-		std::string f1 = filename_absolute_path(filename);
-		std::string f2 = filename_absolute_path(e->src_file);
+		std::string f2 = e->src_abs_path;
```
Comment (GitMensch) [inline libgixpp/TPESQLProcessing.cpp:572]: ... and here (called for each line above) the same filename was converted to an absolute path.

Comment (mridoni) [reply libgixpp/TPESQLProcessing.cpp:572]: Yes, I need to review this, I will get to it after the release of v1.0.10 that is coming in a few days: there will probably be a v1.0.11 soon after, where I hope to reach feature parity for all the database drivers and miscellaneous features (or fixes post-v1.0.10)

**Comment by GitMensch** [comment]
Rechecked, still applies to 1.0.11.

**Comment by mridoni** [comment]
The diffs from this wouldn't apply cleanly, and commit c9491c57801c9a9aebe0d7aa4d261d4371b7ee9a was supposed to make a lot of modifications and additions. I suppose it stems from the fact that the PR was created before a bunch of things were changed, around 1.0.9, I guess. Since I am not really that proficient with git, and to prevent unwanted changes and regressions, I merged this manually into my repository (commit 4a2fbbb13e91df5ad1e0bd3e4ccd072ca722d6cd, which is the "core" of the PR, anyway, and some miscellaneous stuff like copyright dates, etc.).

I will probably (and soon) publish a v1.0.11**a** source-only package, that integrates all the merged PRs and the fixes in my internal repository (except for a solution to #50, that is the next item on the list, but that will require a few days). This is also because I will have less time to work on GixSQL/Gix-IDE in the next weeks so I want to consolidate the work done so far, but without taking the time to prepare and test binary packages.

**Comment by GitMensch** [comment]
You'd see the difference online at https://github.com/mridoni/gix/pull/45/files, this would also be the effect when clicking on "merge" here.
As that is now merged locally already I'm closing this one.

**Comment by mridoni** [comment]
> You'd see the difference online at https://github.com/mridoni/gix/pull/45/files, this would also be the effect when clicking on "merge" here. As that is now merged locally already I'm closing this one.

Yes, I did that, but https://github.com/mridoni/gix/commit/c9491c57801c9a9aebe0d7aa4d261d4371b7ee9a was merging "too much stuff" for me to be comfortable with it. Thanks for the suggestion anyway!

**Pull request closed without merging**